ResearchKit/Common/ORKResultPredicate.h

```diff
@@ -196,6 +319,24 @@ ORK_CLASS_AVAILABLE
  
  @param taskIdentifier      The identifier of the task whose result you want to match. Pass `nil`
                                 to match the ongoing task.
+ @param stepIdentifier      The identifier of the step whose result you want to match. Passing `nil`
+                                assumes the step identifier is the same as the result identifier.
+ @param resultIdentifier    The identifier of the question result you are interested in.
+ @param expectedString      The expected string answer.
+ 
+ @return A result predicate.
+ */
++ (NSPredicate *)predicateForChoiceQuestionResultWithTaskIdentifier:(nullable NSString *)taskIdentifier
+                                                     stepIdentifier:(nullable NSString *)stepIdentifier
+                                                   resultIdentifier:(NSString *)resultIdentifier
+                                                     expectedString:(NSString *)expectedString;
```

Comment:
Good catch.

I can generalize this method to just take an `id<NSCopying, NSCoding, NSObject>` expected value. And keep the other pattern matching method as is (`predicateForChoiceQuestionResultWithResultIdentifier:(NSString *)matchingPatterns:(NSArray<NSString *> *)`) if you want to use regular expressions for the specific case of a text-containing choice question result (which I guess it's a common use case).